task_processing/interfaces/event.py

```diff
@@ -1,13 +1,18 @@
 from pyrsistent import field
+from pyrsistent import m
+from pyrsistent import PMap
 from pyrsistent import PRecord
 
-# TODO: organize and explain these
+from task_processing.interfaces.task_executor import TaskConfig
 
 
 class Event(PRecord):
     # reference to platform-specific event object
     raw = field()
     # is this the last event for a task?
     terminal = field(type=bool, mandatory=True)
+    success = field(type=(bool, type(None)), initial=None)
     # platform-specific event name
     platform_type = field(type=str)
+    task_config = field(type=TaskConfig)
```

Comment:
the reason we need task_config is because retrying logic has to call .run with task_config so id is not enough. we could store id->config map in retrying executor which can be wasteful.

personally i don't have a strong feeling that task_config shouldn't be part of the event structure. there is only so much data that is related to the task, and it's source configuration that it was launched with - one of the most important things about it. it allows for all sorts of heuristics to be implemented in the composable layers without each having to store a crapload of data.